src/main/java/org/jabref/gui/util/ViewModelTableRowFactory.java

```diff
@@ -104,97 +112,128 @@ public ViewModelTableRowFactory<S> withTooltip(Callback<S, String> toTooltip) {
         return this;
     }
 
+    public ViewModelTableRowFactory<S> withPseudoClass(PseudoClass pseudoClass, Callback<S, ObservableValue<Boolean>> toCondition) {
+        this.pseudoClasses.putIfAbsent(pseudoClass, toCondition);
+        return this;
+    }
+
     @Override
     public TableRow<S> call(TableView<S> tableView) {
-        TableRow<S> row = new TableRow<>();
-
-        if (toTooltip != null) {
-            String tooltipText = toTooltip.call(row.getItem());
-            if (StringUtil.isNotBlank(tooltipText)) {
-                row.setTooltip(new Tooltip(tooltipText));
-            }
-        }
-
-        if (onMouseClickedEvent != null) {
-            row.setOnMouseClicked(event -> {
-                if (!row.isEmpty()) {
-                    onMouseClickedEvent.accept(row.getItem(), event);
+        return new TableRow<>() {
+            final List<Subscription> subscriptions = new ArrayList<>();
+
+            @Override
+            protected void updateItem(S item, boolean empty) {
+                super.updateItem(item, empty);
+
+                // Remove previous subscriptions
+                subscriptions.forEach(Subscription::unsubscribe);
+                subscriptions.clear();
+
+                if (contextMenuFactory != null) {
+                    // We only create the context menu when really necessary
+                    setOnContextMenuRequested(event -> {
+                        if (!isEmpty()) {
+                            setContextMenu(contextMenuFactory.apply(item));
+                            getContextMenu().show(this, event.getScreenX(), event.getScreenY());
+                        }
+                        event.consume();
+                    });
+
+                    // Activate context menu if user presses the "context menu" key
+                    tableView.addEventHandler(KeyEvent.KEY_RELEASED, event -> {
```

Comment:
Why is this registering inside the updateItem code? inside the update it's about the cell, this will effectively adda a new event handler for every cell value inside the table 
https://docs.oracle.com/javase/8/javafx/api/javafx/scene/control/Cell.html#updateItem-T-boolean-

Reply:
I just followed https://github.com/JabRef/jabref/blob/main/src/main/java/org/jabref/gui/util/ViewModelTreeTableRowFactory.java#L135 and it the same before the change only the indentation is different. But I agree with you, why are we adding all these events (`onMouseClickedEvent`, `toOnDragDetected`, etc.) to `updateItem`? Shouldn't these events be linked to the row rather than the cell?
@calixtus, do you have any idea on this?

Reply:
Isn't this a method I the rowfactory? Is the rowfactory creating the cells too? Is there another way to add a listener just to the row, not to the cell? Is the row a node on its own?

Sorry, not home right now, I can't look in the docs right now...